File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.py
def solution(a, b, c):
    answer = (a+b+c)*(a*a + b*b + c*c)
    
    if (a == b):
        if(a == c): # 다같은 경우
            answer = (a+b+c)*(a*a + b*b + c*c)*(a*a*a + b*b*b + c*c*c)
    else:
        if(b != c and a != c):
            answer = (a+b+c)
    
    return answer

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.kt
class Solution {
    fun solution(a: Int, b: Int, c: Int): Int {
        var answer: Int = (a+b+c)*(a*a + b*b + c*c) // 두 숫자만 같은 경우
        
        if (a == b){
            if (b == c){ // 모두 같은 경우의 수
                answer *= (a*a*a + b*b*b + c*c*c)
            }
            
            // 아닌 경우는 a = b / c는 따로
        }
        
        else{ //a != b 인 경우
            if (a != c && b != c){ // 모두 같은 경우의 수
                answer = a+b+c 
            }
        }
        
        
        return answer
    }
}